### PR TITLE
Remove size prop from InlineLink

### LINKED ISSUE #446: 🐛 [BUG] - InlineLink does not inherit size
### Describe the bug

Upgrading @primer/react-brand to 0.24.0 and higher causes `InlineLinks` to display as size 200 instead of inheriting their size from their parent element. [For example](https://resources.github.com/learn/pathways/automation/essentials/automated-application-deployment-with-github-actions-and-pages/):

Screenshot with arrows pointing to improperly sized links

I tested @primer/react-brand 0.20.0 and 0.23.0 and the links display as expected:

Screenshot with arrows pointing to properly sized links

### Reproduction steps

1. Create codespace for https://github.com/github/resources-hub
2. npm install @primer/react-brand@0.24.0
3. script/server
4. http://127.0.0.1:3000/learn/pathways/automation/essentials/automated-application-deployment-with-github-actions-and-pages/
5. Links do not inherit size from their parent element

### Expected behavior

1. Create codespace for https://github.com/github/resources-hub
2. npm install @primer/react-brand
3. script/server
4. http://127.0.0.1:3000/learn/pathways/automation/essentials/automated-application-deployment-with-github-actions-and-pages/
5. Links inherit size

### Screenshots

Above

### Browsers

Chrome

### OS

Mac

### PR DESCRIPTION
**Pull request by joseph-lozano**
## Summary

Fixes #446. InlineLinks will no longer take a size prop. This means that they will work in `Heading` and `Text` components which have different size scales.

## List of notable changes:

- **Updated** InlineLinks. They no longer take a `size` prop, but will instead always inherit font-size and line-height. This means that they work inside Heading as well as Text components

## What should reviewers focus on?

Are there any edge cases not considered? Are there any components that InlineLink could be placed in where it would not correctly inherit size?

## Steps to test:

You can view the InlineLink stories in Storybook, or check the diffs in the snapshots.

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

CleanShot 2023-10-23 at 10 19 18@2x

 </td>
<td valign="top">

CleanShot 2023-10-23 at 10 19 36@2x

</td>
</tr>
</table>

### CODE REVIEW

File: .changeset/gold-cars-drop.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@primer/react-brand': minor
+---
+
+**breaking** InlineLink components no longer take a `size` prop, but will now inherit their size in all cases.
+InlineLinks now work inside of `Header` components.
```
Comment (rezrah) [inline .changeset/gold-cars-drop.md:6]: This is great. On breaking changes we usually issue a migration example to make it super obvious how to fix their inevitable type errors.

Could you add this bit in please? 👇 

```diff
+ <InlineLink size="100" />
- <InlineLink /> // size will be inherited from the parent
```

**Review by rezrah — APPROVED** [review]
Some non-blocking feedback. 

Otherwise, looks good. Nice one 🚀

File: packages/react/src/InlineLink/InlineLink.module.css [hunk 2]
```diff
@@ -13,7 +13,9 @@
 
 .InlineLink,
 .InlineLink > span {
+  font-size: inherit;
```
[the remark this answers is not in the document]
Comment (joseph-lozano) [reply packages/react/src/InlineLink/InlineLink.module.css:16]: `InlineLink`s will inherit their font family by default. This regression happens because InlineLinks no longer have an inner `<Text>` tag as part of their implementation. In this story, it is a bare `<InlineLink>`, not wrapped by a `<Text>` tag, so fallback to the default system font.

I don't want to add `font-family: var(--brand-body-fontFamily);`, because then font-family will not not be inherited.

I could wrap the Story in a <Text> tag, but then I think we lose the spirit of what the default story is doing (just showing the bare component).

[the remark this answers is not in the document]
Comment (rezrah) [reply packages/react/src/InlineLink/InlineLink.module.css:16]: Sounds good, thanks for clarifying.
